lib/core/test/unit/Cardano/Wallet/Primitive/Types/StateDeltaSeqSpec.hs

```diff
@@ -0,0 +1,654 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ViewPatterns #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.Wallet.Primitive.Types.StateDeltaSeqSpec
+    ( spec
+    ) where
+
+import Prelude hiding
+    ( seq )
+
+import Cardano.Wallet.Primitive.Types.StateDeltaSeq
+    ( StateDeltaSeq )
+import Data.Function
+    ( (&) )
+import GHC.Generics
+    ( Generic )
+import Safe
+    ( tailMay )
+import Test.Hspec
+    ( Spec, describe, it )
+import Test.QuickCheck
+    ( Arbitrary (..)
+    , CoArbitrary
+    , Fun
+    , Function
+    , Gen
+    , Property
+    , applyFun
+    , arbitraryBoundedEnum
+    , checkCoverage
+    , choose
+    , cover
+    , genericShrink
+    , listOf
+    , property
+    , shrinkMapBy
+    , (===)
+    )
+import Test.QuickCheck.Classes
+    ( bifoldableLaws
+    , bifunctorLaws
+    , eqLaws
+    , foldableLaws
+    , functorLaws
+    , showLaws
+    )
+import Test.Utils.Laws
+    ( testLawsMany )
+
+import qualified Cardano.Wallet.Primitive.Types.StateDeltaSeq as Seq
+import qualified Data.Foldable as F
+import qualified Data.List.NonEmpty as NE
+
+spec :: Spec
+spec = do
+
+    describe "Instances" $ do
+
+        testLawsMany @(StateDeltaSeq TestState TestDelta)
+            [ eqLaws
+            , showLaws
+            ]
+        testLawsMany @(StateDeltaSeq TestState)
+            [ foldableLaws
+            , functorLaws
+            ]
+        testLawsMany @StateDeltaSeq
+            [ bifoldableLaws
+            , bifunctorLaws
+            ]
```

Comment:
Nice!

Reply:
Thanks! I'm very grateful for the existence of the `quickcheck-classes` library.